### PR TITLE
Fix various small mistakes with keyboard tabbing logic - also port to TypeScript

### LINKED ISSUE #1650: Disallow tabbing on "collapse additional textanswer" button
1. Go to the evaluation site for "Game Programming"
2. Don't click anywhere, just press Tab until you are in the form area
3. You _should_ land on the neutral option in the first row, but you land on the leftmost option

If you check where the keyboard focus is right before tabbing onto the wrong element (using `document.activeElement` in the console), it is the "collapse additional textanswer field" button left of the voting buttons (the one with the speech bubble). One shouldn't be able to tab onto the left buttons anyway, so giving them a tabindex of -1 should be the first step to fix this issue.

### PR DESCRIPTION
**Pull request by niklasmohrin**
In addition to porting the file, this closes #1650

### CODE REVIEW

File: evap/static/ts/src/student-vote.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+    if (!(0 <= index && index < row.children.length)) {
+        return;
+    }
+
+    const nextElement = row.children[index] as HTMLElement;
+    nextElement.click();
+}
+
+const studentForm = document.getElementById("student-vote-form") as HTMLElement;
```
[the remark this answers is not in the document]
Comment (niklasmohrin) [reply evap/static/ts/src/student-vote.ts:22]: About the `const` error: I fixed it by using `type="module"` and not `type="text/javascript"` like it was before. I don't know why, but it works :D

About the other comment: I just aborted when either Ctrl of Alt is pressed

**Review by janno42 — COMMENTED** [review]
when submitting with some questions still missing answers, on the reloaded page i cannot start typing a text answer after tabbing somewhere.

**Review by janno42 — COMMENTED** [review]
- after submitting an incomplete form, unanswered questions are shown with the error class adding red borders. navigating with the arrow keys on the answer buttons of a rating question, the error class is not removed, although an answer is selected.
- when tabbing into the voting area, in the first row the first button is selected instead of the center option.

File: evap/static/ts/src/student-vote.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+const selectable: NodeListOf<HTMLElement> = studentForm.querySelectorAll(".tab-selectable");
```
Comment (He3lixxx) [inline evap/static/ts/src/student-vote.ts:23]: `selectables`?

**Review by He3lixxx — APPROVED** [review]

**Review by He3lixxx — APPROVED** [review]

**Review by janno42 — CHANGES_REQUESTED** [review]
after having submitted an incomplete form, the page reloads with all completely answered contributors closed. the card headers are then not tabbable, hence the cards cannot be opened via the keyboard.

**Review by He3lixxx — APPROVED** [review]

**Pull request merged**